In [1]:
import pandas as pd
import numpy as np
import warnings
import statsmodels.api as sm
from statsmodels.formula.api import ols
from scipy.stats import f_oneway

warnings.filterwarnings('ignore')

# Part 1
## Context
Suppose you are working as an analyst in a microprocessor chip manufacturing plant. You have been given the task of analyzing a plasma etching process with respect to changing Power (in Watts) of the plasma beam. Data was collected and provided to you to conduct statistical analysis and check if changing the power of the plasma beam has any effect on the etching rate by the machine. You will conduct ANOVA and check if there is any difference in the mean etching rate for different levels of power. You can find the data anova_lab_data.xlsx file in the files_for_lab folder

In [2]:
data = pd.read_excel('files_for_lab/anova_lab_data.xlsx', sheet_name='data_collected')
data

Power   Etching Rate
0   160 W          5.43
1   180 W          6.24
2   200 W          8.79
3   160 W          5.71
4   180 W          6.71
5   200 W          9.20
6   160 W          6.22
7   180 W          5.98
8   200 W          7.90
9   160 W          6.01
10  180 W          5.66
11  200 W          8.15
12  160 W          5.59
13  180 W          6.60
14  200 W          7.55

In [3]:
data= data.rename(columns={"Power ":"power","Etching Rate":"etching_rate"})
data

power  etching_rate
0   160 W          5.43
1   180 W          6.24
2   200 W          8.79
3   160 W          5.71
4   180 W          6.71
5   200 W          9.20
6   160 W          6.22
7   180 W          5.98
8   200 W          7.90
9   160 W          6.01
10  180 W          5.66
11  200 W          8.15
12  160 W          5.59
13  180 W          6.60
14  200 W          7.55

In [4]:
data.groupby("power").agg(np.mean)

etching_rate
power              
160 W         5.792
180 W         6.238
200 W         8.318

- **State the null hypothesis** <br>
Changing the power of the plasma beam has no effect on the etching rate. The etching rate with 160 W, 180 W and 200 W is equal.

- **State the alternate hypothesis** <br>
Different power of the plasma beam has an effect on the etching rate. The etching rate is different with at least one of the power levels (looks like the etching rate is positively correlated with the power, and the highest rate is with 200 W).

- **What is the significance level**<br>
The level of significance is 0.05. It's the possibility to reject the null hypothesis when it is actually true (false positive).

- **What are the degrees of freedom of model, error terms, and total DoF**<br>
The degrees of freedom are 2 (model), 12 (error terms) and 14 (total).

# Part 2
In this section, use Python to conduct ANOVA. What conclusions can you draw from the experiment and why?

In [5]:
# Using stats model
model = ols('etching_rate ~ C(power)',data=data).fit()
sm.stats.anova_lm(model)

df     sum_sq   mean_sq          F    PR(>F)
C(power)   2.0  18.176653  9.088327  36.878955  0.000008
Residual  12.0   2.957240  0.246437        NaN       NaN

In [14]:
# Using scipy stats

w160 = list(data[data["power"]=="160 W"]["etching_rate"])
w180 = list(data[data["power"]=="180 W"]["etching_rate"])
w200 = list(data[data["power"]=="200 W"]["etching_rate"])

f_oneway(w160, w180, w200)

F_onewayResult(statistic=36.87895470100505, pvalue=7.506584272358903e-06)

The critical level is 3.8853 (http://socr.ucla.edu/Applets.dir/F_Table.html) <br>
F = 36.88 > 3.8853 <br>
Pvalue = 0.000008 < 0.05 <br>
Therefore we reject the null hypothesis. The power of the beam has an effect on the etching rate.